doc/source/reference/arrays.classes.rst

```diff
@@ -625,13 +633,14 @@ As mentioned previously, the flat attribute of ndarray objects returns
 an iterator that will cycle over the entire array in C-style
 contiguous order.
 
->>> for i, val in enumerate(a.flat):
-...     if i%5 == 0: print(i, val)
-0 10
-5 15
-10 20
-15 25
-20 30
+  >>> import numpy as np
```

Comment:
this example doesn't need the import, does it?

Reply:
And please double-check all the imports you're adding are needed.

Reply:
Thanks, I did try my best to make sure I didn't add any unnecessary imports – will make note of this. In this particular example: I think we need it because `a` as an array isn't defined (it's defined in the "Default iteration" heading above it, though. I added `a` from to make the example work in dba775253d6b82d8e565e272ee1d83267264f2b2

Reply:
So this is part of what I was alluding to above with the one import per rst thing, the examples can be interspersed with text but still share a namespace and still make sense to the reader IMO. That means an object declared earlier in the same example is available later without needing to be redefined in every example, which IMO makes the docs harder to read. I guess that makes it harder to make the examples runnable with pyodide?

Reply:
> I guess that makes it harder to make the examples runnable with pyodide?

Yes, in some way. However, there are other directives offered by `jupyterlite-sphinx` such as the [`NotebookLite`](https://jupyterlite-sphinx.readthedocs.io/en/stable/directives/notebooklite.html) directive that can convert an entire page to a runnable example (if it is a MyST-style notebook, for example – and such conversion demands slightly more work). https://github.com/scipy/scipy/pull/20303 and https://github.com/scipy/scipy/pull/21042 are some references for this.

> the examples can be interspersed with text but still share a namespace and still make sense to the reader IMO

For examples that are supposed with the `TryExamples` directive, this could cause problems, because `.. try_examples::` stops the collection of doctest-based code and text once it hits a change in indentation, i.e., after that one needs to add the directive again for another set of code in `>>>`. This is probably okay for the API examples, because the "Examples" rubric is at the end of the docstring with no extra text after it. However, these sections will probably need the imports so if we have to truly make the examples" truly self-contained". I agree that it could probably make following along a little difficult and doing so isn't one of the _primary_ goals of this PR...